addons/point_of_sale/models/pos_session.py

```diff
@@ -300,14 +299,13 @@ def _validate_session(self):
             self._create_picking_at_end_of_session()
         self.with_company(self.company_id)._create_account_move()
         if self.move_id.line_ids:
-            self.move_id.post()
             # Set the uninvoiced orders' state to 'done'
             self.env['pos.order'].search([('session_id', '=', self.id), ('state', '=', 'paid')]).write({'state': 'done'})
         else:
             # The cash register needs to be confirmed for cash diffs
             # made thru cash in/out when sesion is in cash_control.
             if self.config_id.cash_control:
-                self.cash_register_id.button_confirm_bank()
+                self.cash_register_id.button_validate()
```

Comment:
sure it shouldn't be button_validate_or_action?

Reply:
button_confirm_bank -> button_validate
check_confirm_bank -> button_validate_or_action

according to: https://github.com/odoo/odoo/blob/master/addons/account/models/account_bank_statement.py#L370